rpc/client.go

```diff
@@ -104,9 +104,9 @@ func uploadTestsToSPS(ctx context.Context, report *poplar.Report, client *http.C
 	if response.StatusCode != http.StatusOK {
 		body, err := io.ReadAll(response.Body)
 		if err == nil {
-			return errors.Errorf("unexpected status code: %d: %s", response.StatusCode, body)
+			return errors.Wrapf(err, "unexpected status code: %d: %s", response.StatusCode, body)
```

Comment:
I don't think we want to wrap this one though? What happens if you wrap a nil err.